test/pylib/minio_server.py

```diff
@@ -128,12 +127,29 @@ def _anonymous_public_policy(self):
         return {'Statement': statement,
                 'Version': '2012-10-17'}
 
+    async def _is_port_available(self, port: int) -> bool:
+        try:
+            _, writer = await asyncio.open_connection('localhost', port)
+            writer.close()
```

Comment:
Do we have to wait for some timeout here? How quick do we get 'connection refused'?

I'd rather just run the minio server and look at the log in case of non zero exit code, if it says the port is in use, then we need to retry with a different port. This way we won't have any overheads in the main case when the port is free.

Reply:
> Do we have to wait for some timeout here? How quick do we get 'connection refused'?

i am testing using this tiny script
```py
import asyncio

async def foo():
    try:
        await asyncio.open_connection('localhost', 9000)
    except:
        print("===========")

loop = asyncio.get_event_loop()
loop.run_until_complete(foo())
```

```console
$ time python3 /tmp/test.py

real    0m0.044s
user    0m0.036s
sys     0m0.008s
```

connection refusal is not the result of a timeout. the client get the bad news almost immediately. i think it is good enough in our use case.

> I'd rather just run the minio server and look at the log in case of non zero exit code, if it says the port is in use, then we need to retry with a different port. This way we won't have any overheads in the main case when the port is free.

actually, i thought about this. but i hate to parse a human readable output. it's fragile.